app/org/sagebionetworks/bridge/services/UploadService.java

```diff
@@ -289,8 +293,11 @@ public Upload getUpload(@Nonnull String uploadId) {
             return builder.build();
         }).collect(Collectors.toList());
         
-        return new ForwardCursorPagedResourceList<UploadView>(views, list.getOffsetKey(), list.getPageSize())
-                .withFilter("startTime", startTime).withFilter("endTime", endTime);
+        return new ForwardCursorPagedResourceList<UploadView>(views, list.getNextPageOffsetKey())
+                .withRequestParam(OFFSET_KEY, list.getRequestParams().get(OFFSET_KEY))
+                .withRequestParam(PAGE_SIZE, list.getRequestParams().get(PAGE_SIZE))
+                .withRequestParam(START_TIME, startTime)
+                .withRequestParam(END_TIME, endTime);
```

Comment:
I changed this so that the map (whatever it is) is copied entirely, so we don't have to track and test manually copying.